server/handler/assignments.go

```diff
@@ -0,0 +1,71 @@
+package handler
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/src-d/code-annotation/server/repository"
+	"github.com/src-d/code-annotation/server/serializer"
+
+	"github.com/go-chi/chi"
+)
+
+func GetAssignmentsForUserExperiment() RequestProcessFunc {
+	return func(r *http.Request) (resp *serializer.Response, errs []*serializer.Error) {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

also you lose all the profit of go type checking.

consider code:
```go
func(r *http.Request) (resp *serializer.Response, errs []*serializer.Error) {
  ...a lot of code...
  if somecheck {
    resp = postResponse()
  }
  ...a lot of code...
  resp = getResponse()
  return
}
```
code above compiles and produces wrong response.

this signature and always return new value avoids the problem:
```go
func(r *http.Request) (*serializer.Response, []*serializer.Error) {
```